src/sys/ts7200/mod.rs

```diff
@@ -301,7 +223,7 @@ impl Target for Ts7200 {
             PowerState::Halt => {
                 // unlike `run`, we do _not_ want to block the gdb thread waiting
```

Comment:
I've forgotten a lot of context, in light of this change, is this comment still relevant/correct?

Reply:
Good catch, that's a stale comment.

The only issue with switching this to blocking is that sending a interrupt via GDB (i.e: ctrl-c) won't actually trigger until the emulated system is woken up, though that's not too bad.

Reply:
Given that this is only going to run when the cpu put the system in halt state, and so you expect an interrupt soon, it's probably fine, yeah.  Ideally though it would still be nice to be able to pass in the blocking mode, as this makes debugging specifically why your system isn't waking up from halt harder.  But yeah it's probably fine, that's very specific.

Reply:
Oh true, the step method is already parameterized by an argument which changes based on if it's invoked from the main loop or the `gdbstub`, so I could just add another parameter to set the blocking mode.